tests/common/mod.rs

```diff
@@ -0,0 +1,283 @@
+    let mut electrsd_conf = electrsd::Conf::default();
+    let electrsd_dir = tempdir().unwrap();
+    let electrsd_dir_path = electrsd_dir.path().clone().to_path_buf();
+    electrsd_conf.tmpdir = Some(electrsd_dir_path);
+    electrsd_conf.http_enabled = true;
+    electrsd_conf.network = "regtest";
+
+    ElectrsD::with_conf(electrs_exe, &bitcoind, &electrsd_conf).unwrap()
+}
+
+#[cfg(all(target_os = "linux", target_arch = "x86_64"))]
+fn lnd_download_filename() -> String {
+    format!("lnd-linux-amd64-v{}-beta.tar.gz", &LND_VERSION)
+}
+
+#[cfg(all(target_os = "macos", target_arch = "x86_64"))]
+fn lnd_download_filename() -> String {
+    format!("lnd-darwin-arm64-v{}-beta.tar.gz", &LND_VERSION)
+}
+
+#[cfg(all(target_os = "windows"))]
+fn lnd_download_filename() -> String {
+    panic!("Running integration tests on Windows os is not currently supported");
+}
+
+// download_lnd downloads the lnd binary to tmp if it doesn't exist yet. Currently it only downloads binaries
+// compatible with the linux os.
+pub async fn download_lnd() -> TempDir {
+    let temp_dir = tempdir().unwrap();
+    let lnd_dir = &temp_dir.path().to_path_buf();
+    let lnd_releases_endpoint =
+        "https://github.com/lightningnetwork/lnd/releases/download/v0.16.2-beta";
+    let lnd_download_endpoint = format!("{lnd_releases_endpoint}/{}", lnd_download_filename());
+
+    let resp = minreq::get(&lnd_download_endpoint).send().unwrap();
+    assert_eq!(
+        resp.status_code, 200,
+        "url {lnd_download_endpoint} didn't return 200"
+    );
+    let content = resp.as_bytes();
+
+    let decoder = GzDecoder::new(&content[..]);
+    let mut archive = Archive::new(decoder);
+    archive.unpack(lnd_dir).expect("unable to unpack lnd files");
+
+    temp_dir
+}
+
+// LndNode holds the tools we need to interact with a Lightning node.
+#[derive(Debug)]
+pub struct LndNode {
+    address: String,
+    _lnd_exe_dir: TempDir,
+    _lnd_dir_tmp: TempDir,
+    cert_path: String,
+    _handle: Child,
+}
+
+impl LndNode {
+    fn new(bitcoind_connect_params: ConnectParams, lnd_exe_dir: TempDir) -> LndNode {
+        env::set_current_dir(lnd_exe_dir.path().join("lnd-linux-amd64-v0.16.2-beta"))
+            .expect("couldn't set current directory");
+
+        let lnd_dir_binding = Builder::new().prefix("lnd").tempdir().unwrap();
+        let lnd_dir = lnd_dir_binding.path();
+
+        let cert_path = lnd_dir.join("tls.cert").to_str().unwrap().to_string();
+
+        // Have node run on a randomly assigned grpc port. That way, if we run more than one lnd node, they won't
+        // clash.
+        let port = bitcoind::get_available_port().unwrap();
+        let rpc_addr = format!("localhost:{}", port);
+        let args = [
+            format!("--rpclisten={}", rpc_addr),
+            format!("--norest"),
+            format!("--bitcoin.active"),
+            format!("--bitcoin.node=bitcoind"),
+            format!("--bitcoin.regtest"),
```

Comment:
Mainly just because it creates the needed String type! I was thinking of doing String::from() for each, but since format! is used elsewhere figured might as well just use that